tests/test_sed.py

```diff
@@ -316,23 +338,63 @@ def test_SED_init():
         np.testing.assert_raises(TypeError, galsim.SED, spec=lambda w:1.0,
                                  flux_type='bar')
         np.testing.assert_raises(TypeError, galsim.SED, spec=lambda w:1.0)
+        np.testing.assert_raises(ValueError, galsim.SED, spec='wave',
+                                 wave_type=units.Hz, flux_type='2')
+        np.testing.assert_raises(ValueError, galsim.SED, 1.0, 'nm', 'fphotons')
     except ImportError:
         print('The assert_raises tests require nose')
     # These should succeed.
     galsim.SED(spec='wave', wave_type='nm', flux_type='flambda')
     galsim.SED(spec='wave/wave', wave_type='nm', flux_type='flambda')
     galsim.SED(spec=lambda w:1.0, wave_type='nm', flux_type='flambda')
     galsim.SED(spec='1./(wave-700)', wave_type='nm', flux_type='flambda')
+    galsim.SED(spec='wave', wave_type=units.nm, flux_type='flambda')
+    galsim.SED(spec='wave', wave_type=units.Hz, flux_type='flambda')
+    galsim.SED(spec='wave', wave_type=units.Hz, flux_type=units.erg/(units.s*units.nm*units.m**2))
+    galsim.SED(spec='wave', wave_type=units.Hz, flux_type=units.erg/(units.s*units.Hz*units.m**2))
+    galsim.SED(spec='wave', wave_type=units.Hz,
+               flux_type=units.astrophys.photon/(units.s * units.Hz * units.m**2))
+    galsim.SED(spec='wave', wave_type=units.Hz, flux_type='1')
+    galsim.SED(spec='wave', wave_type=units.Hz, flux_type=units.dimensionless_unscaled)
 
     # Also check for invalid calls
     foo = np.arange(10.)+1.
     sed = galsim.SED(galsim.LookupTable(foo,foo), wave_type='nm', flux_type='flambda')
     try:
         np.testing.assert_raises(ValueError, sed, 0.5)
         np.testing.assert_raises(ValueError, sed, 12.0)
+        np.testing.assert_raises(TypeError, galsim.SED, '1', 'nm', units.erg/units.s)
+        np.testing.assert_raises(ValueError, galsim.SED, '1', 'nm', '2')
     except ImportError:
         print('The assert_raises tests require nose')
 
+    # Check a few valid calls for when fast=False
+    sed = galsim.SED(galsim.LookupTable(foo,foo), wave_type='nm', flux_type='flambda',
+                     fast=False)
+    sed(1.5)
+    sed(1.5*units.nm)
+    sed((1.5*units.nm).to(units.Hz, units.spectral()))  # Frequency
+    sed((1.5*units.nm).to(units.erg, units.spectral()))  # Wavelength as an energy
+    sed((1.5*units.nm).to(units.m**-1, units.spectral()))  # Wavenumber in 1/m
+    sed((1.5*units.nm).to(units.radian / units.m, units.spectral()))  # Wavenumber in rad/m
```

Comment:
>Shouldn't we be checking the values of these? Or are we just checking that astropy doesn't raise an exception?

I removed these, which were mostly for my own edification.  I don't think there's much added value in checking astropy.